pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle/ecdhes_aead_composite_encrypt.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package subtle
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/tink/go/subtle/hybrid"
+	"github.com/google/tink/go/subtle/random"
+
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/api"
+)
+
+// ECDHESAEADCompositeEncrypt is an instance of ECDH-ES encryption with Concat KDF
+// and AEAD content encryption
+type ECDHESAEADCompositeEncrypt struct {
+	recPublicKeys []*hybrid.ECPublicKey
+	pointFormat   string
+	encHelper     EncrypterHelper
+}
+
+var _ api.CompositeEncrypt = (*ECDHESAEADCompositeEncrypt)(nil)
+
+// NewECDHESAEADCompositeEncrypt returns ECDH-ES encryption construct with Concat KDF key wrapping
+// and AEAD content encryption
+func NewECDHESAEADCompositeEncrypt(recipientsKeys []*hybrid.ECPublicKey, ptFormat string,
+	encHelper EncrypterHelper) (*ECDHESAEADCompositeEncrypt, error) {
+	var recipients []*hybrid.ECPublicKey
+
+	for _, pub := range recipientsKeys {
+		pubKey := &hybrid.ECPublicKey{
+			Curve: pub.Curve,
+			Point: hybrid.ECPoint{
+				X: pub.Point.X,
+				Y: pub.Point.Y,
+			},
+		}
+
+		recipients = append(recipients, pubKey)
+	}
+
+	return &ECDHESAEADCompositeEncrypt{
+		recPublicKeys: recipients,
+		pointFormat:   ptFormat,
+		encHelper:     encHelper,
+	}, nil
+}
+
+// Encrypt using composite ECDH-ES with a Concat KDF key wrap and AEAD content encryption
+func (e *ECDHESAEADCompositeEncrypt) Encrypt(plaintext, aad []byte) ([]byte, error) {
+	if len(e.recPublicKeys) == 0 {
+		return nil, fmt.Errorf("ECDHESAEADCompositeEncrypt: missing recipients public keys for key wrapping")
+	}
+
+	keySize := e.encHelper.GetSymmetricKeySize()
+	tagSize := e.encHelper.GetTagSize()
+	ivSize := e.encHelper.GetIVSize()
+	cek := random.GetRandomBytes(uint32(keySize))
+
+	var recipientsWK []*RecipientWrappedKey
+
+	for _, rec := range e.recPublicKeys {
+		senderKW := &ECDHESConcatKDFSenderKW{
+			recipientPublicKey: rec,
+			cek:                cek,
+		}
+
+		kek, err := senderKW.wrapKey(A256KWAlg, keySize)
```

Comment:
when we get to chacha, where do we do the additional handling?

Reply:
As far as the [JWE RFC](https://github.com/hyperledger/aries-rfcs/tree/master/features/0334-jwe-envelope#content-encryption-algorithms) is concerned, chacha will be used for Content encryption only, not key wrapping.

Although, if we want to support chacha key wrapping, I believe the logic [here](https://github.com/hyperledger/aries-framework-go/pull/1653/files/c2362a0548b400a4f418d37f925b8242e2afd432#diff-ee10b84659202daf624ebaf5311c1c59R41-R51) will be the same for chacha, the alg name will be different though. To best tested if we want chacha key wrapping. But chacha key wrapping is a first.

Reply:
The DIDComm WG is also allowing for chacha key wrapping.

Reply:
I added a TODO for this.